apps/bfd-db-migrator/src/main/java/gov/cms/bfd/migrator/app/AppConfiguration.java

```diff
@@ -0,0 +1,242 @@
+  }
+
+  /** @see Object#toString() */
+  @Override
+  public String toString() {
+    StringBuilder builder = new StringBuilder();
+    builder.append(", databaseOptions=");
+    builder.append(databaseOptions);
+    builder.append(", metricsOptions=");
+    builder.append(metricOptions);
+    builder.append(", flywayScriptLocationOverride=");
+    builder.append(flywayScriptLocationOverride);
+    return builder.toString();
+  }
+
+  /**
+   * Per <code>/dev/design-decisions-readme.md</code>, this application accepts its configuration
+   * via environment variables. Read those in, and build an {@link AppConfiguration} instance from
+   * them.
+   *
+   * @return the {@link AppConfiguration} instance represented by the configuration provided to this
+   *     application via the environment variables
+   * @throws AppConfigurationException An {@link AppConfigurationException} will be thrown if the
+   *     configuration passed to the application are incomplete or incorrect.
+   */
+  static AppConfiguration readConfigFromEnvironmentVariables() {
+    String databaseUrl = readEnvStringRequired(ENV_VAR_KEY_DATABASE_URL);
+    String databaseUsername = readEnvStringRequired(ENV_VAR_KEY_DATABASE_USERNAME);
+    String databasePassword = readEnvStringRequired(ENV_VAR_KEY_DATABASE_PASSWORD);
+    Optional<String> newRelicMetricKey = readEnvStringOptional(ENV_VAR_NEW_RELIC_METRIC_KEY);
+    Optional<String> newRelicAppName = readEnvStringOptional(ENV_VAR_NEW_RELIC_APP_NAME);
+    Optional<String> newRelicMetricHost = readEnvStringOptional(ENV_VAR_NEW_RELIC_METRIC_HOST);
+    Optional<String> newRelicMetricPath = readEnvStringOptional(ENV_VAR_NEW_RELIC_METRIC_PATH);
+    Optional<Integer> newRelicMetricPeriod = readEnvIntOptional(ENV_VAR_NEW_RELIC_METRIC_PERIOD);
+    Optional<Integer> databaseMaxPoolSize = readEnvIntOptional(ENV_VAR_KEY_DATABASE_MAX_POOL_SIZE);
+    Optional<String> flywayScriptLocationOverride =
+        readEnvStringOptional(ENV_VAR_FLYWAY_SCRIPT_LOCATION);
+    if (databaseMaxPoolSize.isPresent() && databaseMaxPoolSize.get() < 1)
+      throw new AppConfigurationException(
+          String.format(
+              "Invalid value for configuration environment variable '%s': '%s'",
+              ENV_VAR_KEY_DATABASE_MAX_POOL_SIZE, databaseMaxPoolSize));
+
+    Optional<String> hostname;
+    try {
+      hostname = Optional.of(InetAddress.getLocalHost().getHostName());
+    } catch (UnknownHostException e) {
+      hostname = Optional.empty();
+    }
+
+    String flywayScriptLocation = flywayScriptLocationOverride.orElse("");
+
+    MetricOptions metricOptions =
+        new MetricOptions(
+            newRelicMetricKey,
+            newRelicAppName,
+            newRelicMetricHost,
+            newRelicMetricPath,
+            newRelicMetricPeriod,
+            hostname);
+    DatabaseOptions databaseOptions =
+        new DatabaseOptions(
+            databaseUrl, databaseUsername, databasePassword, databaseMaxPoolSize.get());
```

Comment:
this can end up calling get() on an empty optional so needs an .orElse() or other type of logic

Reply:
Good catch, I'll fix this.